misc/python/materialize/cloudtest/app/materialize_application.py

```diff
@@ -161,3 +161,70 @@ def set_environmentd_failpoints(self, failpoints: str) -> None:
         stateful_set.env["FAILPOINTS"] = failpoints
         stateful_set.replace()
         self.wait_for_sql()
+
+    def get_k8s_value(
+        self, selector: str, json_path: str, remove_quotes: bool = True
+    ) -> str:
+        value = self.kubectl(
+            "get",
+            "pods",
+            f"--selector={selector}",
+            "-o",
+            f"jsonpath='{json_path}'",
+        )
+
+        if remove_quotes:
+            value = value.replace("'", "")
+
+        return value
+
+    def get_value_of_pod(
+        self, cluster_id: str, json_path: str, remove_quotes: bool = True
+    ) -> str:
+        return self.get_k8s_value(
+            f"cluster.environmentd.materialize.cloud/cluster-id={cluster_id}",
+            json_path,
+            remove_quotes,
+        )
+
+    def get_label_value_of_pod(
+        self, cluster_id: str, label: str, remove_quotes: bool = True
+    ) -> str:
+        return self.get_value_of_pod(
+            cluster_id, "{.items[*].metadata.labels." + label + "}", remove_quotes
+        )
+
+    def get_node_names_of_pod_by_cluster_name(self, cluster_name: str) -> List[str]:
```

Comment:
`get_cluster_nodes` ?

Reply:
<s>Changed to `get_cluster_nodes_by_cluster_name` because we also have a `get_cluster_nodes_by_cluster_id`</s> No longer after addressing the next comment ;-)

✅